In [1]:
import pandas as pd

In [2]:
df= pd.read_excel('BCG.xlsx')

In [3]:
df.head()

Company  Fiscal Year  Total Revenue(in Billion USD)  \
0      Apple         2024                         391035   
1      Apple         2023                         383285   
2      Apple         2022                         394328   
3  Microsoft         2024                         245122   
4  Microsoft         2023                         211915   

   Net Income(in millions)  Total Assets(in millions)  \
0                    93736                     364980   
1                    96995                     352583   
2                    99803                     352755   
3                    88136                     512163   
4                    72361                     411976   

   Total Liabilities(in millions)  Cash Flow (Operating)(in millions)  
0                          308030                              118254  
1                          290437                              110543  
2                          302083                              122151  
3                          243686                              118548  
4                          205753                               87582

In [11]:
import re
df.columns= [re.sub(r"\s*\(.*?\)", "", col) for col in df.columns]


In [12]:
df.head()

Company  Fiscal Year  Total Revenue  Net Income  Total Assets  \
0      Apple         2024         391035       93736        364980   
1      Apple         2023         383285       96995        352583   
2      Apple         2022         394328       99803        352755   
3  Microsoft         2024         245122       88136        512163   
4  Microsoft         2023         211915       72361        411976   

   Total Liabilities  Cash Flow  
0             308030     118254  
1             290437     110543  
2             302083     122151  
3             243686     118548  
4             205753      87582

In [18]:
df['revenue growth %']=df.groupby(['Company'])['Total Revenue'].pct_change()*100
df['net income growth %']=df.groupby(['Company'])['Net Income'].pct_change()*100

In [19]:
df.head(10)

Company  Fiscal Year  Total Revenue  Net Income  Total Assets  \
0      Apple         2024         391035       93736        364980   
1      Apple         2023         383285       96995        352583   
2      Apple         2022         394328       99803        352755   
3  Microsoft         2024         245122       88136        512163   
4  Microsoft         2023         211915       72361        411976   
5  Microsoft         2022         198270       72738        364840   
6      Tesla         2024          97690        7153        122070   
7      Tesla         2023          96773       14974        106618   
8      Tesla         2022          81462       12587         82338   

   Total Liabilities  Cash Flow  revenue growth %  net income growth %  
0             308030     118254               NaN                  NaN  
1             290437     110543         -1.981920             3.476786  
2             302083     122151          2.881146             2.894995  
3             243686     118548               NaN                  NaN  
4             205753      87582        -13.547132           -17.898475  
5             198298      89035         -6.438902             0.520999  
6              48390      14923               NaN                  NaN  
7              43009      13256         -0.938684           109.338739  
8              36440      14724        -15.821562           -15.940964

<h2>Revenue Growth</h2>
Apple had a decrese in revenue by 1.98 % in 2024 compared to 2023, but there was a rise of 2.88% in revenue in 2023 compared to 2022. 
Microsoft had diped 6.43% in 2023 since 2022, and also extra 13.5% in 2024 from 2023. Revenue is decreasing more each year.
Tesle had 15.8% fall in the revenue in 2023, but it improved and had fall of only 0.9% in 2024 since 2023.
Overall all there is a decresing tread in all the companies.



<h2>Net Income Growth</h2>
Unlike revenue, Net income has increased by 2.8% in 2023 and by 3.47% in 2024 for Apple. But its not the same case for Microsoft, although there was a rise of 0.52% of net income in year 2023 compared to 2022, it dropped in 2024 with a sharp 15%. Its totally the opposite case for Tesla, although there was 15% drop in net income in 2023, Tesla turned it around with 109.3% total growth in net income in year 2024.
